src/caldavclient.cpp

```diff
@@ -423,6 +384,62 @@ void CalDavClient::start()
         return;
     }
 
+    // After removal of libsocialcache dependency, remove all data and force clean sync.
+    //    - remove any notebooks associated with this account
+    //    - remove any notebooks associated with non-existent accounts
+    //    - set mFirstSync = true.
+    {
+        QString accountIdString = iProfile.key(Buteo::KEY_ACCOUNT_ID);
+        int accountId = accountIdString.toInt();
+        QString settingsFileName = QString::fromLatin1("/home/nemo/.local/share/system/privileged/Sync/caldav.ini");
+        QSettings settingsFile(settingsFileName, QSettings::IniFormat);
+        bool alreadyClean = settingsFile.value(QStringLiteral("%1-cleaned").arg(accountId), QVariant::fromValue<bool>(false)).toBool();
+        if (!alreadyClean) {
+            // first, delete any data associated with this account, so this sync will be a clean sync.
+            LOG_WARNING("Deleting caldav notebooks associated with this account:" << accountId << "due to clean sync");
+            deleteNotebooksForAccount(accountId, mCalendar, mStorage);
+            // now delete notebooks for non-existent accounts.
+            LOG_WARNING("Deleting caldav notebooks associated with nonexistent accounts due to clean sync");
+            // a) find out which accounts are associated with each of our notebooks.
+            QList<int> notebookAccountIds;
+            mKCal::Notebook::List allNotebooks = mStorage->notebooks();
+            Q_FOREACH (mKCal::Notebook::Ptr nb, allNotebooks) {
+                QString nbAccount = nb->account();
+                if (!nbAccount.isEmpty() && nb->pluginName().contains(QStringLiteral("caldav"))) {
+                    // caldav notebook->account() values used to be like: "55-/user/calendars/someCalendar"
+                    int indexOfHyphen = nbAccount.indexOf('-');
+                    if (indexOfHyphen > 0) {
+                        // this is a caldav notebook from pre 1.1.6, since it uses "accountId-remoteServerPath" form
```

Comment:
1.1.6 and earlier rather, though references to system version numbers in this library might be misleading